public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/validation.ts

```diff
@@ -16,6 +16,23 @@ interface ParseError {
   node: SyntaxNode;
 }
 
+/**
+ * Simplified version of `validateQuery` to determine if a given query is valid or not.
+ * Grafana variables are errors to the parser, so it is ideal to call this function with the
+ * interpolated query.
+ * @alpha
+ */
+export function isValidQuery(interpolatedQuery: string): boolean {
+  return validateQuery(interpolatedQuery, interpolatedQuery, interpolatedQuery.split('\\n')) === false ? true : false;
+}
```

Comment:
I might oversee something, but what's the better way to check if a query is valid now? Do we need both cases?

```ts
isValidQuery(this.interpolateString(query, placeHolderScopedVars))
```

or 

```ts
!isQueryWithError(this.interpolateString(query, placeHolderScopedVars))
```

Reply:
`isValidQuery` is the one. `isQueryWithError` looks for error nodes. I actually thought of naming it `isQueryWithErrorNode` to be more explicit.

Reply:
We should probably also use it in the Datasource too, no?

```
async getDataSamples(query: LokiQuery): Promise<DataFrame[]> {
    // Currently works only for logs sample
    if (isQueryWithError(query.expr) || !isLogsQuery(query.expr)) {
      return [];
    }
    // ...
```

Reply:
I mean, if the query has a variable we will not request samples.

Reply:
> isValidQuery is the one. isQueryWithError looks for error nodes.

But what's the logical difference in running both on an interpolated query expression? Both should return the same, right? Well, `isValidQuery(interpolatedExpr) !== isQueryWithError(interpolatedExpr)` would be always true.